extensions/notebook/src/extension.ts

```diff
@@ -53,6 +65,38 @@ async function openNotebook(): Promise<void> {
 	}
 }
 
+async function runActiveCell(): Promise<void> {
+	try {
+		let notebook = sqlops.nb.activeNotebookEditor;
+		if (notebook) {
+			await notebook.runCell();
+		} else {
+			throw new Error(noNotebookVisible);
+		}
+	} catch (err) {
+		vscode.window.showErrorMessage(err);
+	}
+}
+
+async function addCell(cellType: sqlops.nb.CellType): Promise<void> {
+	try {
+		let notebook = sqlops.nb.activeNotebookEditor;
+		if (notebook) {
+			await notebook.edit((editBuilder: sqlops.nb.NotebookEditorEdit) => {
+				// TODO should prompt and handle cell placement
+				editBuilder.insertCell({
+					cell_type: cellType,
+					source: ''
+				});
+			});
+		} else {
+			throw new Error(noNotebookVisible);
```

Comment:
Will user see this exception? If I am the user, I prefer not getting this error when no notebook is visible

Reply:
Well in theory they shouldn't as the context key means the action shouldn't show unless it's visible. I tested this just now and confirmed that's the case. If we changed the logic though, we do need some way for users to see why their action didn't do anything.